packages/ui-patterns/CommandMenu/prepackaged/ai/queryAi.ts

```diff
@@ -0,0 +1,69 @@
+import type OpenAI from 'openai'
+import { SSE } from 'sse.js'
+
+import { type Message } from './utils'
+import { BASE_PATH } from '../shared/constants'
+
+/**
+ * Perform a one-off query to AI based on a snapshot of messages
+ */
+export function queryAi(messages: Message[], timeout = 0) {
+  return new Promise<string>((resolve, reject) => {
+    const eventSource = new SSE(`${BASE_PATH}/api/ai/docs`, {
+      headers: {
+        apikey: process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY ?? '',
+        Authorization: `Bearer ${process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY}`,
+        'Content-Type': 'application/json',
+      },
+      payload: JSON.stringify({
+        messages: messages.map(({ role, content }) => ({ role, content })),
+      }),
+    })
+
+    let timeoutId: number | undefined
+
+    function handleError<T>(err: T) {
+      if (timeoutId) {
+        clearTimeout(timeoutId)
+      }
+      console.error(err)
+      reject(err)
+    }
+
+    if (timeout > 0) {
+      timeoutId = window.setTimeout(() => {
+        handleError(new Error('AI query timed out'))
+      }, timeout)
+    }
+
+    let answer = ''
+
+    eventSource.addEventListener('error', handleError)
+    eventSource.addEventListener('message', (e: MessageEvent) => {
+      try {
+        if (e.data === '[DONE]') {
+          if (timeoutId) {
+            clearTimeout(timeoutId)
+          }
+          resolve(answer)
+          return
+        }
+
+        const completionChunk: OpenAI.Chat.Completions.ChatCompletionChunk = JSON.parse(e.data)
+        const [
+          {
+            delta: { content },
+          },
+        ] = completionChunk.choices
+
+        if (content) {
+          answer += content
+        }
+      } catch (err) {
+        handleError(err)
+      }
+    })
+
+    eventSource.stream()
```

Comment:
Not sure this is the right place to leave this comment on, but it seems like the ai answers don't come in in a stream fashion in the docs cmdk menu. But rather show up only when the answer is complete. See screen recording.

https://github.com/user-attachments/assets/cb34a45e-76cb-4bc8-9760-e682a8b0734e